libs/seiscomp/io/recordstream/concurrent.cpp

```diff
@@ -297,9 +297,22 @@ Record *Concurrent::next() {
 	}
 
 	try {
-		return _queue.pop();
+		Record * rec =  _queue.pop();
+		if ( rec) {
+			return rec;
+		}
+
+		if ( _queue.size() > 0 ) {
```

Comment:
That must never happen. We push `nullptr` after the last thread terminated. No way that there are more items in the queue.

Reply:
Anyway, we can keep it for safety and monitor the logs.

Reply:
I added that check to avoid regressions in the future. However I was tempted to drop it too.